src/components/select/select.ts

```diff
@@ -0,0 +1,377 @@
+    this.selectItem(this.allItems[prev], true);
+  }
+
+  protected selectFirstActiveItem() {
+    const items = this.allItems.filter((i) => !i.disabled);
+    const index = this.allItems.indexOf(items[0]);
+    const item = this.allItems[index];
+
+    if (item.value !== this.value) {
+      this.selectItem(item, true);
+    }
+  }
+
+  protected selectLastActiveItem() {
+    const items = this.allItems.filter((i) => !i.disabled);
+    const index = this.allItems.indexOf(items[items.length - 1]);
+    const item = this.allItems[index];
+
+    if (item.value !== this.value) {
+      this.selectItem(item, true);
+    }
+  }
+
+  protected selectItemByKey(event: KeyboardEvent): void {
+    if (
+      !event ||
+      !event.key ||
+      event.key.length > 1 ||
+      event.key === ' ' ||
+      event.key === 'spacebar'
+    ) {
+      // ignore longer keys ('Alt', 'ArrowDown', etc) AND spacebar (used of open/close)
+      return;
+    }
+
+    const currentTime = Date.now();
+
+    if (currentTime - this.lastKeyTime > 500) {
+      this.searchTerm = '';
+    }
+
+    this.searchTerm += event.key;
+    this.lastKeyTime = currentTime;
+
+    const item = this.allItems
+      .filter((i) => !i.disabled)
+      .find((i) =>
+        i.textContent
+          ?.trim()
+          .toLowerCase()
+          .startsWith(this.searchTerm.toLowerCase())
+      );
+    if (item) this.selectItem(item);
+  }
+
+  protected handleInputKeyboardEvents(event: KeyboardEvent) {
+    event.stopPropagation();
+    const key = event.key.toLowerCase();
+
+    if (
+      event.altKey &&
+      (key === 'arrowup' ||
+        key === 'arrowdown' ||
+        key === 'down' ||
+        key === 'up')
+    ) {
+      this.toggle();
+      return;
+    }
+
+    if (!this.open) {
+      switch (key) {
+        case 'space':
+        case 'spacebar':
+        case ' ':
+        case 'enter':
+          event.preventDefault();
+          this.show();
+          return;
+        case 'arrowdown':
+        case 'down':
+        case 'arrowright':
+        case 'right':
+          event.preventDefault();
+          this.selectNext();
+          return;
+        case 'arrowup':
+        case 'up':
+        case 'arrowleft':
+        case 'left':
+          event.preventDefault();
+          this.selectPrev();
+          return;
+        case 'home':
+          event.preventDefault();
+          this.selectFirstActiveItem();
+          return;
+        case 'end':
+          event.preventDefault();
+          this.selectLastActiveItem();
+          return;
+        default:
+          break;
+      }
+    } else if (key === 'tab' || (event.shiftKey && key === 'tab')) {
+      this.hide();
+    }
+
+    this.handleKeyDown(event);
+    this.selectItemByKey(event);
+  }
+
+  protected renderInnerSlots(el: HTMLElement, slot: string) {
+    el.setAttribute('slot', slot);
+    return html`${el.cloneNode(true)}`;
+  }
+
+  protected override render() {
+    return html`
+      <igc-input
```

Comment:
There's quire a bit of ARIA missing, such as the role of the control and the linkage to the listbox and its active item. Then again, currently the `igc-input` will not transfer such props to the native input, right?